benchmark/wildtime_benchmarks/data_generation_arxiv.py

```diff
@@ -44,39 +48,89 @@ def __init__(self,  data_dir):
         self.path = data_dir
 
     def store_data(self, store_all_data: bool, add_final_dummy_year: bool):
-        for year in tqdm(self._dataset):
+        # create directories
+        if not os.path.exists(self.path):
+            os.mkdir(self.path)
+
+        train_dir = os.path.join(self.path, "train")
+        os.makedirs(train_dir, exist_ok=True)
+
+        if store_all_data:
+            test_dir = os.path.join(self.path, "test")
+            os.makedirs(test_dir, exist_ok=True)
+
+        stats = {}
+
+        for year in self._dataset:
             # for simplicity, instead of using years we map each day to a year from 1970
             year_timestamp = create_timestamp(year=1970, month=1, day=year-2006)
-            year_rows = []
 
-            splits = [0, 1] if store_all_data else [0]
-            for split in splits:
+            def get_one_split(split: int) -> list[str]:
+                rows = []
                 for i in range(len(self._dataset[year][split]["title"])):
                     text = self._dataset[year][split]["title"][i].replace("\n", " ")
                     label = self._dataset[year][split]["category"][i]
                     csv_row = f"{text}\t{label}"
-                    year_rows.append(csv_row)
+                    rows.append(csv_row)
+                return rows
 
-            # store the year file
-            text_file = os.path.join(self.path, f"{year}.csv")
-            with open(text_file, "w", encoding="utf-8") as f:
-                f.write("\n".join(year_rows))
+            train_year_rows = get_one_split(0)
```

Comment:
This more like a general question: If I understand correctly, the pre-supplied arXiv data already has a test/train split, and we use that, correct? Have we actually validated that this split is balanced, in some sense? What I mean is that whoever created the dataset might not have created a balanced test split, it could contain many samples from e.g. 2007 and 0 samples from 2009 or something like that. 

If this is the case, we might need to merge the train+test data supplied from the arxiv dataset, and then manually create a test split per year. Actually, I thought this would happen in this PR. It's okay to use the pre-defined split, but we should validate that the split behaves as expected

Reply:
Yes the pre-supplied dataset already has a test/train split, and we use it (also I think the wildtime paper uses it as well). It has a ~10:1 ratio (train vs test), so it's quite balanced

Reply:
fyi
![image](https://github.com/eth-easl/modyn/assets/100585941/7ee6cf3c-eb27-4ce2-a538-cf6d4c70a6ec)

Reply:
Ok, thank you, those numbers look reasonable!